pkg/healthcheck/healthcheck.go

```diff
@@ -1854,6 +1869,43 @@ func (hc *HealthChecker) CheckNamespace(ctx context.Context, namespace string, s
 	return nil
 }
 
+func (hc *HealthChecker) checkClusterNetworks(ctx context.Context) error {
+	nodes, err := hc.kubeAPI.GetNodes(ctx)
+	if err != nil {
+		return err
+	}
+	var podCIDRs []string
+	for _, node := range nodes {
+		podCIDRs = append(podCIDRs, node.Spec.PodCIDR)
+	}
+	clusterNetworks := strings.Split(hc.linkerdConfig.ClusterNetworks, ",")
+	clusterIPNets := make([]*net.IPNet, len(clusterNetworks))
+	for i, clusterNetwork := range clusterNetworks {
+		_, ipNet, err := net.ParseCIDR(clusterNetwork)
+		if err != nil {
+			return err
+		}
+		clusterIPNets[i] = ipNet
+	}
+	for _, podCIDR := range podCIDRs {
+		podIP, _, err := net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+		exists := false
+		for _, clusterIPNet := range clusterIPNets {
+			if clusterIPNet.Contains(podIP) {
+				exists = true
+				break
+			}
+		}
```

Comment:
I think it's better not to short-circuit on the first error, but instead accumulate in a list all the PodCIDRs that don't match the clusterNetworks config, and present them in the error string.